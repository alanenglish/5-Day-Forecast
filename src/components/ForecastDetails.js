import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

const ForecastDetails = ({ forecast }) => (
  <li className="list-inline-item forecast-item">
    <div className="forecast-item__date">
      {moment(forecast.applicable_date).format('dddd')}
    </div>
    <img
      className="forecast-item__img"
      alt={forecast.weather_state_name}
      src={`https://www.metaweather.com/static/img/weather/${forecast.weather_state_abbr}.svg`}
    />
    <div className="forecast-item__description" >
      {forecast.weather_state_name}
    </div>
    <div className="forecast-item__hi-low" >
      Hi: <span className="hi-temp">{Math.floor((forecast.max_temp * (9 / 5)) + 32)}</span>
      Low: <span className="low-temp">{Math.floor((forecast.min_temp * (9 / 5)) + 32)}</span>
    </div>
  </li>
);

ForecastDetails.propTypes = {
  forecast: PropTypes.object
};

ForecastDetails.defaultProps = {
  forecast: null
};

export default ForecastDetails;
