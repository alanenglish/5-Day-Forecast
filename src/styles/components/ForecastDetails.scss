.forecast-item {
  background-color: #F1F1F1;
  border-radius: 4px;
  box-shadow: 0 6px 4px -4px #ADADAD;
  height: 325px;
  margin: 0px 10px;
  text-align: center;
  vertical-align:top;
  width: 255px;

  .forecast-item__date {
    font-family: 'Oswald', sans-serif;
    font-size: 26px;
    font-weight: bold;
    letter-spacing: 1px;
    margin: 10px 0;
  }

  .forecast-item__img {
    background-color: #B0CAEC;
    height: 175px;
    margin-top: 5px;
    width: 100%;
  }

  .forecast-item__description{
    font-family: 'Oswald', sans-serif;
    font-size: 24px;
    height: 40px;
    margin-top: 5px;
    width: inherit;
  }

  .forecast-item__hi-low {
    font-family: 'Oswald', sans-serif;
    font-size: 18px;
    font-weight: lighter;
    margin-left: 5px;
    .hi-temp {
      color: #BE4949;
      margin-right: 5px;
    }
    .low-temp {
      color: #5597ED;
    }
  }
}

@media only screen and (max-width: 768px) {
    /* For mobile phones: */
    .forecast-item {
      height: 225px;
      margin: 0px 5px;
      width: 135px;
      .forecast-item__date {
        font-size: 20px;
        margin: 5px 0;
      }
      .forecast-item__img {
        height: 115px;
        margin-top: 5px;
        width: 100%;
      }
      .forecast-item__description {
        font-size: 18px;
        height: 30px;
        margin-top: 3px;
      }
      .forecast-item__hi-low {
        font-size: 12px;
        margin-left: 3px;
        .hi-temp {
          margin-right: 3px;
        }
      }
    }
}

@media only screen and (max-width: 370px) {
    /* For mobile phones: */

}
