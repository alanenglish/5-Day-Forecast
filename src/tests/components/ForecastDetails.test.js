import React from 'react';
import { shallow } from 'enzyme';
import ForecastDetails from '../../components/ForecastDetails';
/* eslint-env jest */

let forecast;

beforeEach(() => {
  forecast = {
    id: 5120283761442816,
    weather_state_name: 'Light Rain',
    weather_state_abbr: 'lr',
    applicable_date: '2018-12-08',
    min_temp: 7.9325,
    max_temp: 12.5
  };
});

test('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ForecastDetails forecast={forecast} />);
  expect(wrapper).toMatchSnapshot();
});
