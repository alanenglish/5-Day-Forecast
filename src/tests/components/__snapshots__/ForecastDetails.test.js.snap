// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<li
  className="list-inline-item forecast-item"
>
  <div
    className="forecast-item__date"
  >
    Saturday
  </div>
  <img
    alt="Light Rain"
    className="forecast-item__img"
    src="https://www.metaweather.com/static/img/weather/lr.svg"
  />
  <div
    className="forecast-item__description"
  >
    Light Rain
  </div>
  <div
    className="forecast-item__hi-low"
  >
    Hi: 
    <span
      className="hi-temp"
    >
      54
    </span>
    Low: 
    <span
      className="low-temp"
    >
      46
    </span>
  </div>
</li>
`;
